internal/bminventory/inventory.go

```diff
@@ -1523,31 +1523,49 @@ func (b *bareMetalInventory) DownloadClusterFiles(ctx context.Context, params in
 }
 
 func (b *bareMetalInventory) DownloadClusterKubeconfig(ctx context.Context, params installer.DownloadClusterKubeconfigParams) middleware.Responder {
+	if apiErr := b.checkFileForDownload(ctx, params.ClusterID.String(), kubeconfig); apiErr != nil {
+		return apiErr
+	}
+
+	respBody, contentLength, err := b.s3Client.Download(ctx, fmt.Sprintf("%s/%s", params.ClusterID, kubeconfig))
+	if err != nil {
+		return installer.NewDownloadClusterKubeconfigConflict().
+			WithPayload(common.GenerateError(http.StatusConflict, errors.Wrap(err, "failed to download kubeconfig")))
+	}
+	return filemiddleware.NewResponder(installer.NewDownloadClusterKubeconfigOK().WithPayload(respBody), kubeconfig, contentLength)
+}
+
+func (b *bareMetalInventory) checkFileForDownload(ctx context.Context, clusterID, fileName string) *common.ApiErrorResponse {
```

Comment:
Usually we don't return `*common.ApiErrorResponse` in internal functions, just return an error that can be of type `*common.ApiErrorResponse` and then when you return to the user you can call to `return common.GenerateErrorResponder(err)`